tests/python/pants_test/logging/test_native_engine_logging.py

```diff
@@ -13,7 +15,10 @@ def test_native_logging(self):
     pants_run = self.run_pants([
       "-linfo", "list", "3rdparty::"
     ])
-    self.assertNotRegexpMatches(pants_run.stderr_data, "DEBUG] Launching \\d+ root")
+    if PY3:
+      self.assertNotRegex(pants_run.stderr_data, "DEBUG] Launching \\d+ root")
+    else:
+      self.assertNotRegexpMatches(pants_run.stderr_data, "DEBUG] Launching \\d+ root")
```

Comment:
I don't like all this duplication of test parameters, but because this is temporary, I think it's less confusing to keep it than to do something like
```python
assert_regex_commands = [pants_run.stderr_data, "DEBUG] Launching \\d+ root"]
if PY3:
  self.assertNotRegex(*assert_regex_commands)
else:
  self.assertNotRegexpMatches(*assert_regex_commands)
```

Reply:
Given there are a few occurrences, can we make an alias ourselves like
```python
def assertRegexHelper():
    if PY3:
       return self.assertRegex
    else:
       return self.assertRegexMatches
```

Reply:
I think I like that change. I'll add to this PR an example of what that will look like, and we can decide if we want to proceed.

If we factor out `assertNotRegex`, I think we should also factor out `assertRegex`. Even though `assertRegex` vs `assertRegexpMatches` doesn't have this same issue, it would be confusing I think to use `assertRegexpMatches` everywhere but then require using our custom `assertNotRegex` everywhere. This would change about 30 additional lines.

Reply:
Ah I used `assertRegex` just as an example. It looks like only `assertNotRegex` in the scope, so only refactoring `assertNotRegex` would be okay.

Reply:
@wisechengyi thoughts on this setup? https://github.com/pantsbuild/pants/pull/6723/commits/822b99bb00861beedfbef42d2f2226c088ffa1e6

While we don't need to have `assertRegex` extracted out as well, I think we should for consistency with `assertNotRegex`. It would be confusing to me to have to remember in which case you use what, rather than the rule "if it's an assertion with regex matching, use our compat layer."

Reply:
Okay great! I'll apply this change to the remaining instances of `assert*RegexpMatches`. 

Going to wait on an additional review before merging—I could see people not liking it. If you know any reviewers who would have thoughts on this, feel free to add them as reviewers please.